pennylane/templates/subroutines/double_excitation_unitary.py

```diff
@@ -508,45 +473,70 @@ def circuit(weight, wires1=None, wires2=None):
 
     """
 
-    ##############
-    # Input checks
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weight, wires1=None, wires2=None, do_queue=True):
+
+        if len(wires1) < 2:
+            raise ValueError(
+                "expected at least two wires representing the occupied orbitals; "
+                "got {}".format(len(wires1))
+            )
+        if len(wires2) < 2:
+            raise ValueError(
+                "expected at least two wires representing the unoccupied orbitals; "
+                "got {}".format(len(wires2))
+            )
+
+        shape = qml.math.shape(weight)
+        if shape != ():
+            raise ValueError(f"Weight must be a scalar; got shape {shape}.")
+
+        self.wires1 = list(wires1)
+        self.wires2 = list(wires2)
+        wires = wires1 + wires2
```

Comment:
Here I am trying to parse to wires as late as possible (i.e. in the parent function called by `super`). This is a result of the insight that parsing to `Wires` is costly...